src/wasm/wasm-type.cpp

```diff
@@ -65,6 +71,8 @@ struct TypeInfo {
 
   bool isNullable() const { return kind == RefKind && ref.nullable; }
 
+  bool getCanonical(Type& out) const;
```

Comment:
the meaning of the bool result isn't immediately obvious by the name - worth a comment i think

Reply:
Does it do anything else than those two examples?

What I'm looking for is the purpose of the method. "Simpler" in what sense? (`none` is simpler than anything :wink: ) And "respresented" in what sense? (if we can turn a singleton tuple into its sole value, that implies the tuple itself doesn't matter, but it's hard for me to guess at a generalized rule for that.)

Also, it sounds like this is unrelated to canonicalization or recursive types? If so, could it be `getSimpler`?

Reply:
Simpler as in how having Type::funcref is simpler than having a RefKind TypeInfo pointing to a BasicKind HeapTypeInfo that in turn contains HeapType::any. This is crucial for canonicalization and used to be part of `Store<TypeInfo>::canonicalize`. The reason this PR splits them out into separate methods is that the same logic needs to be applied in `FiniteShapeHasher` and `FiniteShapeEquator` (to fix canonicalization bugs) and it's much safer to use the same implementation in all locations than to duplicate the logic.

Reply:
I see, thanks.